pyqmc/linemin.py

```diff
@@ -168,60 +159,43 @@ def line_minimization(
     # Attributes for linemin
     attr = dict(max_iterations=max_iterations, npts=npts, steprange=steprange)
 
-    def gradient_energy_function(x, coords):
-        set_wf_params(wf, x, pgrad_acc)
-        df, coords = pyqmc.mc.vmc(
+    x0 = pgrad_acc.transform.serialize_parameters(wf.parameters)
+
+    df = []
+    # Gradient descent cycles
+    for it in range(max_iterations):
+
+        set_wf_params(wf, x0, pgrad_acc)
+        df_vmc, coords = pyqmc.mc.vmc(
             wf,
             coords,
             accumulators={"pgrad": pgrad_acc},
             client=client,
             npartitions=npartitions,
             **vmcoptions,
         )
-        en = np.real(np.mean(df["pgradtotal"], axis=0))
-        en_err = np.std(df["pgradtotal"], axis=0) / np.sqrt(df["pgradtotal"].shape[0])
-        sigma = np.std(df["pgradtotal"], axis=0) * np.sqrt(np.mean(df["nconfig"]))
-        dpH = np.mean(df["pgraddpH"], axis=0)
-        dp = np.mean(df["pgraddppsi"], axis=0)
-        dpdp = np.mean(df["pgraddpidpj"], axis=0)
-        grad = 2 * np.real(dpH - en * dp)
-        Sij = np.real(dpdp - np.einsum("i,j->ij", dp, dp))
-
-        if np.any(np.isnan(grad)):
-            for nm, quant in {"dpH": dpH, "dp": dp, "en": en}.items():
-                print(nm, quant)
-            raise ValueError("NaN detected in derivatives")
-
-        return coords, grad, Sij, en, en_err, sigma
 
-    x0 = pgrad_acc.transform.serialize_parameters(wf.parameters)
+        data = {}
+        for k in ['pgradtotal','pgraddpH','pgraddppsi','pgraddpidpj']:
```

Comment:
This doesn't work; if we need other things averaged then this won't do it. Should loop over keys() from pgrad_acc.